palette/src/num.rs

```diff
@@ -27,6 +27,14 @@ pub trait Real {
     /// Create a number from an `f64` value, mainly for converting constants.
     #[must_use]
     fn from_f64(n: f64) -> Self;
+
+    /// Returns the negative infinity (−∞).
+    #[must_use]
+    fn neg_inf() -> Self;
+
+    /// Returns the infinity (∞).
+    #[must_use]
+    fn inf() -> Self;
```

Comment:
If these are just for the `Oklab` limits, I'm more willing to drop the limits.

Reply:
It's also the upper limit for the chroma of Oklch (technically they are also for CIELab and CIELch -- ["The a* and b* axes are unbounded, and depending on the reference white they can easily exceed ±150 to cover the human gamut. Nevertheless, software implementations often clamp these values for practical reasons."](https://en.wikipedia.org/wiki/CIELAB_color_space) ). The failed test in CI is exactly about the chroma of Oklch. A last-minute change I made and forgot to rerun the test for.


The doc of `Real` says "Numbers that belong to the real number set." How is that not float specific? Do you plan to include complex numbers? Or special kinds of real numbers that are finite only? In the later case, a trait like the following would make sense
```rust
/// Methods for the finite limits of the numbers.
pub trait Limits{
  /// Returns a finite number, for which `x <= Self::max_value()` holds for all comparable `x`.
  fn max_value() -> Self;

  /// Returns a finite number, for which `x >= Self::min_value()` holds for all comparable `x`.
  fn min_value() -> Self;
}
```

Reply:
Oklch and Oklab are more or less just different coordinate systems in the same space, so I suppose what goes for one goes for the other. Unless there's some definition that says otherwise. I think CIELab has a bit of an exception in that each axis is expected to fit in an `i8` when rounded. CIELch is a bit more difficult to nail down, but the radii are based on the CIELab square.

> The doc of Real says "Numbers that belong to the real number set." How is that not float specific? Do you plan to include complex numbers?

I'm referring to the mathematical real number set, which doesn't include infinity as far as I know. But it's less about number sets and more about widening the set of allowed types to anything that can be used as a substitute for floats. Not all of them has a representation of infinity, so this particular trait is very minimal.

A `Limits` trait could be useful but I prefer to first question why an unbounded color space has bounds. 🤔